### ISSUE TITLE
Training with Official Script Yields Low-Quality Results and High FID

### ISSUE THREAD

**Issue by runjie-yan** [zelaki/ReDi#5]
Hi, thank you for the great work on this project!

I trained the model using the official script with the following command:

```bash
torchrun --nnodes 1 --nproc_per_node 8 train_redi.py \
  --model "SiT-XL/2" \
  --feature-path "./data/imagenet-256-processed" \
  --pca-rank 8 \
  --pca-model-path ./pcs/dino_pca_model.pth
```

The model was trained for 1400 epochs (350,000 steps). However, it only produces low-quality generations (see examples below), and the reported FID score is much worse than expected. I guess there might be some misconfigurations in the released training scripts.

The test results in the `./fids` folder are:

> Computing evaluations...
> Inception Score: 337.853759765625
> FID: 82.22543859825191
> sFID: 38.99878307343408
> Precision: 0.1662
> Recall: 0.54172

To verify this, I tested the `zelaki/SiT-ReDi-XL-2` model from Hugging Face and was able to match the reported performance. (I suspect there might be an issue with IS computation, but FID seems reliable.)

Test results using the Hugging Face model:

> Computing evaluations...
> Inception Score: 337.853759765625
> FID: 1.6233654727982412
> sFID: 4.647570684489779
> Precision: 0.641
> Recall: 0.78182

One key difference I noticed is that the Hugging Face SiT model must be loaded with `learn_sigma=True`, whereas the released training script defaults to `learn_sigma=False`. This discrepancy may be contributing to the performance gap.

Could you please confirm whether this is expected, and if there are any other settings or training configurations that should be adjusted to match the reported results?

Thank you again—any clarification or help would be greatly appreciated!

<img width="256" height="256" alt="Image" src="https://github.com/user-attachments/assets/013eef11-dea4-404a-8df6-0308722e8c6d" />
<img width="256" height="256" alt="Image" src="https://github.com/user-attachments/assets/194ab978-e004-4d81-8900-bb4063a5c04e" />
<img width="256" height="256" alt="Image" src="https://github.com/user-attachments/assets/3fb775e2-4383-4deb-8e61-b4a90653cf76" />

**Comment by tyhuang98** [comment]
I also have the problem. It seems to be related to the guidance setup.

**Comment by zelaki** [comment]
This is weird since we have cross-validated the results in two different environments. Note that learn_sigma is just for compatibility with DiT (which uses DDPM instead of flow matching), and it doesn't affect training (you can set it to False). Did you follow the preprocessing protocol of REPA exactly?

**Comment by runjie-yan** [comment]
> This is weird since we have cross-validated the results in two different environments. Note that learn_sigma is just for compatibility with DiT (which uses DDPM instead of flow matching), and it doesn't affect training (you can set it to False). Did you follow the preprocessing protocol of REPA exactly?

Yes, I believe I have followed the data preprocessing pipeline correctly. Could you please share the exact versions of the packages used in your test environments?

Also, here is a snippet of the training log from my run—could you check if the loss values are consistent with your results?

Thanks a lot for your help!

The first few lines:
> [[34m2025-07-12 09:19:42[0m] Experiment directory created at results/002-SiT-XL-2-Linear-velocitydino-pca-8
> [[34m2025-07-12 09:19:57[0m] SiT Parameters: 675,184,944
> [[34m2025-07-12 09:20:05[0m] Dataset contains 64,148 images (/mnt/pfs/users/yanrunjie/official-redi/data/imagenet-256-processed)
> [[34m2025-07-12 09:20:06[0m] using MLP layer as FFN
> [[34m2025-07-12 09:20:08[0m] Training for 1400 epochs...
> [[34m2025-07-12 09:20:08[0m] Beginning epoch 0...
> [[34m2025-07-12 09:21:06[0m] (step=0000100) Train Loss: 8.4433, Train Steps/Sec: 1.64
> [[34m2025-07-12 09:22:03[0m] (step=0000200) Train Loss: 4.3753, Train Steps/Sec: 1.75
> [[34m2025-07-12 09:22:32[0m] Beginning epoch 1...

The last few lines:
> [[34m2025-07-14 17:51:45[0m] Beginning epoch 1399...
> [[34m2025-07-14 17:52:14[0m] (step=0349800) Train Loss: 1.0651, Train Steps/Sec: 1.73
> [[34m2025-07-14 17:53:12[0m] (step=0349900) Train Loss: 1.0730, Train Steps/Sec: 1.74
> [[34m2025-07-14 17:54:09[0m] (step=0350000) Train Loss: 1.0610, Train Steps/Sec: 1.74
> [[34m2025-07-14 17:54:17[0m] Saved checkpoint to results/002-SiT-XL-2-Linear-velocitydino-pca-8/checkpoints/0350000.pt
> [[34m2025-07-14 17:54:18[0m] Done!
